datacube/drivers/netcdf/driver.py

```diff
@@ -39,15 +42,26 @@ def format(self):
     def uri_scheme(self):
         return PROTOCOL
 
-    def write_dataset_to_storage(self, dataset, filename,
+    def mk_uri(self, file_path, storage_config):
+        """
+        Constructs a uri from the file_path and storage config.
+        resource.
+        """
+        driver_alias = storage_config['driver']
```

Comment:
For this driver, I think it should always return `normalise_path(file_path).as_uri()` and simply ignore `storage_config` part.